pkg/api/db.go

```diff
@@ -187,7 +187,9 @@ func CreateVolumeSnapshotDBEntry(ctx *c.Context, in *model.VolumeSnapshotSpec) (
 }
 
 func DeleteVolumeSnapshotDBEntry(ctx *c.Context, in *model.VolumeSnapshotSpec) error {
-	if in.Status != model.VolumeSnapAvailable {
+	invalidStatus := []string{model.VolumeSnapAvailable, model.VolumeSnapError,
```

Comment:
If one snapshot is in model.VolumeSnapAvailable and model.VolumeSnapError, model.VolumeSnapErrorDeleting, we could delete it.

Reply:
And when one snapshot in creatint status, we should not delete it.

Reply:
What do you want to say?